src/qoi/include/grins/rayfire_mesh.h

```diff
@@ -0,0 +1,126 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// GRINS - General Reacting Incompressible Navier-Stokes
+//
+// Copyright (C) 2014-2016 Paul T. Bauman, Roy H. Stogner
+// Copyright (C) 2010-2013 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+
+#ifndef GRINS_RAYFIRE_MESH_H
+#define GRINS_RAYFIRE_MESH_H
+
+// libMesh
+#include "libmesh/quadrature.h"
+#include "libmesh/mesh.h"
+
+// GRINS
+#include "grins/qoi_base.h"
+#include "grins/variable_name_defaults.h"
+
+namespace GRINS
+{
+    
+  //! RayfireMesh
+  /*!
+  This class creates a 1D mesh of EDGE2 elements to allow
+  for integration of functions along a line across the domain
+  */
+  class RayfireMesh
+  {
+  public:
+    
+    //! 2D Constructor
+    /*! 
+    The init() function must be called after the constructor
+    @param origin Origin point (x,y) of the rayfire on mesh boundary
+    @param theta Spherical polar angle (in radians)
+    */
+    RayfireMesh(libMesh::Point& origin, libMesh::Real theta);
+    
+    //! 3D Constructor
+    /*! 
+    The init() function must be called after the constructor
+    @param origin Origin point (x,y,z) of the rayfire on mesh boundary
+    @param theta  Spherical polar angle (in radians)
+    @param phi    Spherical azimuthal angle (in radians)
+    */
+    RayfireMesh(libMesh::Point& origin, libMesh::Real theta, libMesh::Real phi);
+    
+    //! Initialization
+    /*! 
+    This function performs the rayfire and assembles the 1D mesh 
+    @param mesh_base Reference to the main mesh
+    */
+    void init(const libMesh::MeshBase& mesh_base);
+    
+    /*! 
+    This function takes in an elem_id on the main mesh and returns an elem from the 1D rayfire mesh 
+    @param elem_id The ID of the elem on the main mesh
+    */
+    const libMesh::Elem* translate(const libMesh::dof_id_type elem_id);
```

Comment:
This is not a good function name for this. `translate` to me means translating an element or something. I'd do something like `map_main_elem_to_1d_elem()`. Also, how does this function behave if there's no underlying 1D element? The documentation should tell the user.